### PR TITLE
[pkg/ottl] adding code for double converter

### LINKED ISSUE #22056: [pkg/ottl]: Add Double converter
### Component(s)

pkg/ottl

### Is your feature request related to a problem? Please describe.

There's currently an `Int` converter, but no obvious way to force a value to be a `Double`. If you have an attribute that's a string representation of a double, it'd be nice to just parse it the way you currently for string representations of integers with `Int`.

### Describe the solution you'd like

Add a `Double` converter that can convert `Double`, `Int`, `String` values.

### Describe alternatives you've considered

The best I've come up with is going via JSON:
```
ParseJSON(Concat(["{\"value\":", attributes["some_attr"], "}"], ""))["value"]
```
which does in fact work for Double, Int, and String.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gord02**
**Description:** Adding a Double converter to pkg/ottl

**Link to tracking Issue:** 

closes #22056

**Testing:** I added similar tests as in func_int.go

### CODE REVIEW

**Review by TylerHelmuth — COMMENTED** [review]
@gord02 this looks good.  Also update`ottl/ottlfuncs/functions.go` with the new factory and add a `ottl/ottlfuncs/README` entry.

**Comment by TylerHelmuth** [comment]
@gord02 when you're ready for a full review please take the PR out of draft.

**Comment by gord02** [comment]
Hello @TylerHelmuth. Will this PR need a changelog entry or can it be skipped?

**Comment by TylerHelmuth** [comment]
@gord02 it will need an `enhancement` changelog entry for `pkg/ottl`

File: pkg/ottl/ottlfuncs/README.md [hunk 1]
```diff
@@ -351,6 +352,29 @@ Examples:
 
 - `ConvertCase(metric.name, "snake")`
 
+### Double
+
+The `Double` Converter converts an inputted `value` into a double.
+
+The returned type is double64.
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:359]:
```suggestion
The returned type is float64.
```

File: pkg/ottl/ottlfuncs/functions.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ func converters[K any]() []ottl.Factory[K] {
 		NewConcatFactory[K](),
 		NewConvertCaseFactory[K](),
 		NewDurationFactory[K](),
+		NewDoubleFactory[K](),
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/functions.go:40]: Please move this above `NewDurationFactory` to keep the list alphabetical.

**Comment by gord02** [comment]
Does anyone have any suggestions on how to resolve the failing 'integration-tests' check?

**Review by TylerHelmuth — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**